### DISCUSSION TITLE
How to add custom document when user first signin and get them

### DISCUSSION THREAD

**Discussion by IRediTOTO** [nextauthjs/next-auth#576] [Help]
**Your question**
Hi, I just start use next-auth. It look so cool but I feel I can't control it. :( 

I am trying to save custom data to users collection. I am trying to

- Save a custom data to users collection, ex: `wallet:0` for first time login
- Then all data of that user pipe into `req` to use for query more easy (Like passport). Of course I can query users again in middleware but It look so silly :( 

I found the Adapter in [document ](https://next-auth.js.org/tutorials/creating-a-database-adapter#example-code)but I don't understand how to use it... and look like don't have a example project.

* [x ] Found the documentation helpful
* [ x] Found documentation but was incomplete
* [ x] Could not find relevant documentation
* [x ] Found the example project helpful
* [x ] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment]
I would suggest you to you the mongodb driver for node or mongoose and do what operations you want using the provided callback functions: https://next-auth.js.org/configuration/callbacks.
